### PR TITLE
add cmake(install by sdk manager)

### PR DESCRIPTION
**Pull request by LeonDevLifeLog**

### CODE REVIEW

**Comment by master-bob** [comment]
That's cool. I didn't know sdkmanager could add cmake.

Is it correct to assume that some projects need older versions of cmake? I.e. some need 3.6 but will not work with 3.22?

**Comment by LeonDevLifeLog** [comment]
> 

Yes, The cmake version can be specified in the build.gradle file.

The cmake installed by apt is outdated.

About #124 , I will update later.

File: Dockerfile [hunk 1]
```diff
@@ -194,6 +194,14 @@ RUN echo "emulator" && \
     . /etc/jdk.env && \
     yes | $ANDROID_SDK_MANAGER "emulator" > /dev/null
 
+RUN echo "build cmake" && \
+    . /etc/jdk.env && \
+    yes | $ANDROID_SDK_MANAGER \
+        "cmake;3.22.1" \
+        "cmake;3.18.1" \
+        "cmake;3.10.2.4988404" \
+        "cmake;3.6.4111459" > /dev/null
```
Comment (mingchen) [inline Dockerfile:203]: Do we really need multiple version of `cmake`? Maybe just install the latest is enough.

Comment (master-bob) [reply Dockerfile:203]: I guess it depends on how gradle handles conflicts with the installed cmake and a specified version in the `build.gradle` file.

**Comment by hossamnasser938** [comment]
Why this has been closed. We need it. I have a project with react-native-vision-camera and the pipeline fails because cMake 3.22.1 is missing

**Comment by master-bob** [comment]
> Why this has been closed? We need it. I have a project with react-native-vision-camera and the pipeline fails because cMake 3.22.1 is missing

It was closed because it was stale, meaning that no updates had been made to the ticket. In more detail, after this was opened #124 was merged which was a major change to how this project was built. As such the original change was no longer valid with the updated Dockerfile.

Nonetheless with the change to [Ubuntu 22.04 LTS](https://packages.ubuntu.com/jammy/cmake), #164, cmake 3.22.1 is the default.

**Comment by hossamnasser938** [comment]
@master-bob Thanks for the clarification. Then what I can understand is if I use `mingc/android-build-box:latest`, I should expect cMake 3.22.1 to be installed, right?

Another question please: I do have a pipeline that works fine locally but fails when using the latest of this image on Jenkins while building react-native-vision-camera with this error "Could not find Ninja on PATH or in SDK CMake bin folders". Do you have any idea what's wrong?

**Comment by master-bob** [comment]
> @master-bob Thanks for the clarification. Then what I can understand is if I use `mingc/android-build-box:latest`, I should expect cMake 3.22.1 to be installed, right?

Latest has 3.22.1 installed.

> Another question please: I do have a pipeline that works fine locally but fails when using the latest of this image on Jenkins while building react-native-vision-camera with this error "Could not find Ninja on PATH or in SDK CMake bin folders". Do you have any idea what's wrong?

This is off-topic, but the issue is that this "Ninja" application is not installed *or* it is installed in a location not in PATH. So the requirements are not being met. PATH is all the locations the OS looks for when trying to run a program. For more detail I suggest you search for it.

**Pull request closed without merging**